plugins/module_utils/_api_data.py

```diff
@@ -1184,14 +1184,14 @@ def join_path(path):
         fully_understood=True,
         primary_keys=('public-key', 'interface'),
         fields={
-            'allowed-address': KeyInfo(),
+            'allowed-address': KeyInfo(required=True),
             'comment': KeyInfo(can_disable=True, remove_value=''),
             'disabled': KeyInfo(default=False),
-            'endpoint-address': KeyInfo(),
+            'endpoint-address': KeyInfo(default=''),
             'endpoint-port': KeyInfo(default=0),
             'interface': KeyInfo(),
-            'persistent-keepalive': KeyInfo(),
-            'preshared-key': KeyInfo(),
+            'persistent-keepalive': KeyInfo(can_disable=True, remove_value=0),
+            'preshared-key': KeyInfo(can_disable=True, remove_value=''),
             'public-key': KeyInfo(),
         },
```

Comment:
CC @cosandr

Reply:
LGTM, causes no changes for my config and `allowed-address` is indeed required.